src/components/Alert/Alert.module.css

```diff
@@ -0,0 +1,131 @@
+/**
+ * Copyright 2024 Mia srl
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+.alert {
+  border-width: 0;
+  border-radius: 0;
+  border-left-width: 40px;
+  padding: var(--spacing-padding-lg, 16px);
+  gap: var(--spacing-gap-lg, 16px);
+  align-items: center;
+
+  background-color: var(--palette-purple-100, #EDE7FF);
+  border-color: var(--palette-purple-500, #986DF1);
+
+  & > svg {
+    position: absolute;
+    align-self: center;
+    left: calc(var(--spacing-gap-sm, 8px) - 40px);
+
+    color: var(--palette-purple-100, #EDE7FF) !important;
+  }
+
+  &:global(.mia-platform-alert-with-description) :global(.mia-platform-alert-message) {
+    margin-bottom: var(--spacing-padding-xs, 4px) !important;
+  }
+
+  & :global(.mia-platform-alert-close-icon) {
+    color: var(--palette-action-secondary-contrastText, #636363);
+  }
+
+  & .titleContainer svg {
+    color:  var(--palette-purple-500, #986DF1) !important;
+  }
+
+}
+
+.compressed {
+  border-left-width: 0;
+  padding: 0 16px;
+}
+
+.titleContainer {
+  display: flex;
+  flex-direction: row;
+  align-items: center;
+  gap: var(--spacing-gap-xs, 4px);
+}
+
+.compressed {
+  border-left-width: 0;
+  border-radius: var(--shape-border-radius-lg, 8px);
+  padding: var(--spacing-padding-md, 12px);
+
+  &:global(.mia-platform-alert-with-description) :global(.mia-platform-alert-message) {
+    padding-bottom: var(--spacing-padding-sm, 8px);
+  }
+
+  & .title {
+    font-size: var(--typography-bodySBold-fontSize, 14px);
+    line-height: var(--typography-bodySBold-lineHeight, 16px);
+    font-weight: var(--typography-bodySBold-fontWeight, 600);
+    color: var(--typography-bodySBold-color, #636363);
+  }
+
+}
+
+.title {
+  font-size: var(--typography-bodyMBold-fontSize, 16px);
+  line-height: var(--typography-bodyMBold-lineHeight, 24px);
+  font-weight: var(--typography-bodyMBold-fontWeight, 600);
+  color: var(--typography-bodyMBold-color, #636363);
+}
+
+.description {
+  font-size: var(--typography-bodyS-fontSize, 14px);
+  line-height: var(--typography-bodyS-lineHeight, 16px);
+  font-weight: var(--typography-bodyS-fontWeight, 400);
+  color: var(--typography-bodyS-color, #636363);
+}
+
+.content {
```

Comment:
can this be explained with a comment on this choice?

Reply:
The problem i noticed is the typography component in the design-system uses a lot of !important in their classes, so it becomes hard to modify the style from the extern (it shouldn't be the case anyway, but for nested components can be a problem, ie. the CheckboxLabel > Checkbox > CheckboxGroup, where the style was the same of the BodyS component but the lineheight was different causing problems in spacings).

Additionally, using the css this way you don't have to write additional codes in the component, like:
if (isCompressed) {
  titleComponent = BodyS
} else {
  titleComponent = BodyM
}

Reply:
EDIT: i've seen the libary exposes also a component BodyX so the switch could be on the prop "size" instead, but using the css imo has always the benefit to limit the coding to the logic, and not to the styling